parsl/monitoring/monitoring.py

```diff
@@ -527,40 +523,47 @@ def router_starter(comm_q: "queue.Queue[Union[Tuple[int, int], str]]",
     router.logger.info("End of router_starter")
 
 
+def send_first_message(try_id: int,
+                       task_id: int,
+                       monitoring_hub_url: str,
+                       run_id: str) -> None:
+    import platform
+
+    radio = UDPRadio(monitoring_hub_url,
+                     source_id=task_id)
+
+    msg = {'run_id': run_id,
+           'try_id': try_id,
+           'task_id': task_id,
+           'hostname': platform.node(),
+           'first_msg': True,
+           'timestamp': datetime.datetime.now()
+    }
+    radio.send(msg)
+    return
+
+
 def monitor(pid: int,
             try_id: int,
             task_id: int,
             monitoring_hub_url: str,
             run_id: str,
             logging_level: int = logging.INFO,
-            sleep_dur: float = 10,
-            first_message: bool = False) -> None:
+            sleep_dur: float = 10) -> None:
     """Internal
     Monitors the Parsl task's resources by pointing psutil to the task's pid and watching it and its children.
     """
+    import logging
     import platform
+    import psutil
     import time
 
     radio = UDPRadio(monitoring_hub_url,
                      source_id=task_id)
 
-    if first_message:
-        msg = {'run_id': run_id,
-               'try_id': try_id,
-               'task_id': task_id,
-               'hostname': platform.node(),
-               'first_msg': first_message,
-               'timestamp': datetime.datetime.now()
-        }
-        radio.send(msg)
-        return
-
-    import psutil
-    import logging
-
     format_string = "%(asctime)s.%(msecs)03d %(name)s:%(lineno)d [%(levelname)s]  %(message)s"
     logging.basicConfig(filename='{logbase}/monitor.{task_id}.{pid}.log'.format(
-        logbase="/tmp", task_id=task_id, pid=pid), level=logging_level, format=format_string)
+        logbase="/tmp", task_id=task_id, pid=pid), level=logging.DEBUG, format=format_string)
```

Comment:
Is it intentionally changed to debug logging here? if yes, maybe should also remove logging_level, as it is not used anywhere now.

Reply:
no. i'll change it back